src/main/java/trinity/play2learn/backend/admin/year/services/YearRegisterService.java

```diff
@@ -0,0 +1,52 @@
+package trinity.play2learn.backend.admin.year.services;
+
+import org.springframework.stereotype.Service;
+import org.springframework.validation.BindingResult;
+
+import trinity.play2learn.backend.admin.year.dtos.YearRequestDto;
+import trinity.play2learn.backend.admin.year.dtos.YearResponseDto;
+import trinity.play2learn.backend.admin.year.mappers.YearMapper;
+import trinity.play2learn.backend.admin.year.models.Year;
+import trinity.play2learn.backend.admin.year.repositories.IYearRepository;
+import trinity.play2learn.backend.admin.year.services.interfaces.IYearExistService;
+import trinity.play2learn.backend.admin.year.services.interfaces.IYearRegisterService;
+import trinity.play2learn.backend.configs.exceptions.BadRequestException;
+
+/**
+ * Servicio para la lógica de negocio relacionada con el caso de uso 7, registrar un año académico.
+ * Encapsula validaciones y operaciones sobre el modelo Year.
+ */
+@Service
+public class YearRegisterService implements IYearRegisterService {
+
+    private IYearExistService  yearExistService;
+    private IYearRepository yearRepository;
+
+    public YearRegisterService(IYearExistService yearExistService, IYearRepository yearRepository) {
+        this.yearExistService = yearExistService;
+        this.yearRepository = yearRepository;
+    }
+
+    /**
+     * Crea un nuevo año si no existe uno con el mismo nombre.
+     *
+     * @param YearRequestDto.
+     * @return YearResponseDto.
+     * @throws BadRequestException si ya existe un año con ese nombre.
+     * @throws BadRequestException si el nombre tiene caracteres especiales.
+     * @throws BadRequestException si el nombre supera los 50 caracteres.
+     * @throws BadRequestException si el nombre es vacio.
+     */
+    @Override
+    public YearResponseDto cu7RegisterYear (YearRequestDto yearRequestDto, BindingResult result) {
+        if (result.hasFieldErrors()) {
+            throw new BadRequestException("An error ocurred.", result.getFieldErrors().stream().map(err -> err.getField() + ": " + err.getDefaultMessage()).toList());
+        }
```

Comment:
Entiendo que acá estás tratando de manejar los errores que ocurren por las restricciones que tienen los DTO. Esto no es necesario: jakarta lo hace por defecto. Las validaciones de jakarta lanzan una `IllegalArgumentException` (creo) con status 400 si las validaaciones no se cumplen. Sacando esto, podés evitar meter el `BindingResult`.

```suggestion
    public YearResponseDto cu7RegisterYear (YearRequestDto yearRequestDto) {
```